plugin/ipam/cni.go

```diff
@@ -37,7 +38,15 @@ func (i *Ipam) Allocate(args *skel.CmdArgs) (types.Result, error) {
 	}
 	var ipnet *net.IPNet
 
-	if conf.Subnet == "" {
+	if len(conf.IPs) == 1 {
+		ip := conf.IPs[0]
+		if ip.Version == "4" {
+			ipnet = &ip.Address
+			err = i.weave.ClaimIP(containerID, ipnet, false)
+		} else {
+			return nil, errors.New("Not Implemented")
```

Comment:
Please include the thing that is not implemented in the error message, don't assume it will always be what you think it is, or that the person reading the error message can figure that out.